src/metax_api/services/auth_service.py

```diff
@@ -20,29 +20,34 @@ def get_user_projects(request):
         """
         Fetches users file projects from local file and from
         token. On local file values must be a list of strings.
+        Projects are cached to request.user.user_projects to increase performance.
         """
+
+        if hasattr(request.user, 'user_projects'):
+            return request.user.user_projects
+
         user_projects = AuthService.extract_file_projects_from_token(request.user.token)
         username = request.user.token.get('CSCUserName', '')
+        file_projects = None
 
         try:
             with open(settings.ADDITIONAL_USER_PROJECTS_PATH, 'r') as file:
                 file_projects = json.load(file)
         except FileNotFoundError:
-            _logger.info("No local file for user projects")
-            return user_projects
+            msg = "No local file for user projects"
         except Exception as e:
-            _logger.error(e)
-            return user_projects
+            msg = e
 
-        try:
-            if isinstance(file_projects[username], list) and isinstance(file_projects[username][0], str):
-                for project in file_projects[username]:
-                    user_projects.add(project)
-            else:
-                _logger.error("Projects on file are not list of strings")
-        except:
+        if file_projects is None:
+            _logger.info(msg) if isinstance(msg, str) else _logger.error(msg)
```

Comment:
Why not log these messages where they happen (inside the ``except`` blocks)? And then handle the rest of the elif stuff like

````
if file_projects:
    if not file_projects.get(username, False):
        ...
````
or such